### ISSUE TITLE
Starting Problems

### ISSUE THREAD

**Issue by Archaeoscan** [jwilczek-dotcom/DACORD#3]
When running Dacord, i get the following code in RStudio:
Lade nötiges Paket: shiny

Listening on http://127.0.0.1:3333
The legacy packages maptools, rgdal, and rgeos, underpinning this package
will retire shortly. Please refer to R-spatial evolution reports on
https://r-spatial.org/r/2023/05/15/evolution4.html for details.
This package is now running under evolution status 0 
Scalable Robust Estimators with High Breakdown Point (version 1.7-3)

Attache Paket: ‘pracma’

Das folgende Objekt ist maskiert ‘package:rrcov’:

    sqrtm

Attache Paket: ‘igraph’

Die folgenden Objekte sind maskiert von ‘package:stats’:

    decompose, spectrum

Das folgende Objekt ist maskiert ‘package:base’:

    union

Warning: Error in &&: 'length = 168926' in coercion to 'logical(1)'
  1: shiny::runApp

I can load a model, but manual pre-orientation doesn't work as it should and the browser window gets grey...

Can anybody help? Im a total rookie to Rstudio and coding.

**Comment by bernb** [comment]
I am not the author, but the issue originates in some indirect dependency of DACORD. See [here](https://github.com/geomorphR/geomorph/issues/43) for more details. This more or less fully breaks the manual preorientation feature. As a workaround, you could try to change line 1121 in `functions.R` from
```
pts <- digit.fixed.lite(mesh,fixed = 3,ptsize = 2,center=F)                      # newly added
```
to
```
pts <- digit.fixed.lite(mesh,fixed = 3,ptsize = 25,center=F)                      # newly added
```
The model will appear black, which is an intentional side effect and it will be harder to recognize the shape of the model. Let me know if this doesn't work, there might be second workaround.

**Comment by Archaeoscan** [comment]
Thank you very much! I will try this as soon as i have the tim to fiddle with it.
But what you are telling is that skipping manual preorientation (and orientate only fully automatic) should keep everything going?

**Comment by bernb** [comment]
Yes, the problem is related to the manual preorientation feature and automatic preorientation should work without errors.

**Comment by Archaeoscan** [comment]
neither your code change nor the automatic preorientation worked: the browser window got grey an nothing happened.

This is what RStudio said:
Warning: Error in &&: 'length = 168926' in coercion to 'logical(1)'
  1: shiny::runApp
[1] "Parallel started"
[1] "Parallel finished in  10 secs."
Warning in cbind(n0, ni[i, ]) :
  number of rows of result is not a multiple of vector length (arg 2)
Warning: Error in optimx.check: Cannot evaluate function at initial parameters
  1: shiny::runApp
Warning: Error in &&: 'length = 121794' in coercion to 'logical(1)'
  1: shiny::runApp
Warning: Error in &&: 'length = 318180' in coercion to 'logical(1)'
  1: shiny::runApp
[1] "Parallel started"
[1] "Parallel finished in  10 secs."
Warning in cbind(n0, ni[i, ]) :
  number of rows of result is not a multiple of vector length (arg 2)
Warning: Error in optimx.check: Cannot evaluate function at initial parameters
  1: shiny::runApp

**Comment by bernb** [comment]
Could you provide a sample file for which the problem occurs? I will try to replicate the problem.

**Comment by Archaeoscan** [comment]
I tried with several 3d models als .ply, Unfortunately github doesn't allow .ply as attachments

**Comment by bernb** [comment]
It does work for my models, so I assume the problem is somehow related to the models you are using. If you don't provide an example file (via wetransfer, dropbox, onedrive or other means) I won't be able to help you.

**Comment by Archaeoscan** [comment]
I didn't see an option to send you a personal message, so here is a link to one of my models that failed: https://we.tl/t-iRMAsnqxkj
Thanks for taking a look!

**Comment by bernb** [comment]
I tried with the file you provided but automatic preorientation does finish (unsuccessfully) without an error. There are however still some things we can do to isolate the problem:
1. Very important: Double check that the same error occurs with the very same file you provided
2. Provide the information given by typing `sessionInfo()` into the R console (bottom of the RStudio window where the error is shown)

The error for automatic preorientationi indicates a problem with the normals of the models. The same error occurred for me some time ago when I tried to use a model without any calculated normals. Your model however does include calculated normals and I don't see anything that would produce the error.
